mitodl/mitxpro
# revert: certificate revisions prior to August 8 2022

#### Pre-Flight checklist


#### What are the relevant tickets?
#2439 

#### What's this PR do?
- Reverts the certificate page revisions for certificates generated prior to August 8, 2022
- This change is very specific to date so you will see hard coded. date in this
- Also adds a reverse script for the certificate migrations in case anything goes wrong and we have to undo the migration so it will try to revert to the things that were set before bases on the production migration run time

#### How should this be manually tested?
1. make sure to have a reasonable set of data to test this on
2. Make sure to have certificates created with the date prior to `August 8` (this means `created_on` in CourseRunCertificates is prior to `August 8 2022`)
3.  Make sure to have PageRevisions prior to `August 8 2022` (this means `created-at` in the PageRevision model is prior to August 8)
4. Redo Step# 2,3 but this time with dates after `August 8 2022`
5. Also make sure that you have multiple revisions for the dame certificates available for dates before and after August 8, 2022
6. When you apply this migration you need to make sure that only certificates prior to `August 8, 2022` are changed with revisions associated with the latest revision as of that Date.

#### Where should the reviewer start?
Create a dataset of mixed dates prior to August 8 2022 and after that date with revisions

#### Any background context you want to provide?
We migrated the data when we added Data Migrations for Page Revisions in https://github.com/mitodl/mitxpro/pull/2416

#### What GIF best describes this PR or how it makes you feel?
![documents-here-there](https://media.giphy.com/media/xiAqCzbB3eZvG/giphy.gif)
